### PR TITLE
Mamoru Sniffer for Light and Full node

### PR DESCRIPTION
**Pull request by gofmanaa**
- add Mamoru Sniffer for Snap and Full node

### CODE REVIEW

File: mamoru/feeder.go [hunk 1]
```diff
@@ -8,6 +8,6 @@ import (
 type Feeder interface {
 	FeedBlock(*types.Block) evm_types.Block
 	FeedTransactions(*types.Block, types.Receipts) []evm_types.Transaction
-	FeedCalTraces([]*TxTraceResult, uint64) []evm_types.CallTrace
 	FeedEvents(types.Receipts) []evm_types.Event
+	FeedCalTraces([]*CallFrame, uint64) []evm_types.CallTrace
```
Comment (sokyrko) [inline mamoru/feeder.go]: `FeedCallTraces`

File: mamoru/tracer.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+		defer func() {
+			if r := recover(); r != nil {
+				log.Error("Mamoru Sniffer Recovered:", r)
+			}
+		}()
+		var err error
+		sniffer, err = mamoru_sniffer.Connect()
+		if err != nil {
+			log.Error("Mamoru Sniffer connect error:", err)
+			panic(err)
```
Comment (sokyrko) [inline mamoru/tracer.go]: What the reason to panic here and recover right after? What happens if we just remove `panic(err)` ?

**Review by sokyrko — APPROVED** [review]

File: mamoru/feeder.go (hunk 1 above)
Comment (gofmanaa) [reply mamoru/feeder.go]: Thank you

File: mamoru/tracer.go (hunk 2 above)
Comment (gofmanaa) [reply mamoru/tracer.go]: You are right, panic is not needed, the code does not even reach it and the recovery does not work out

**Pull request merged**